### DISCUSSION TITLE
QB not populate relation

### DISCUSSION THREAD

**Discussion by vinverdy** [mikro-orm/mikro-orm#5938] [General]
### Describe the bug

relation not loaded, when using leftJoinAndSelect, unexpectedly using leftJoin is fine

### Reproduction

```ts
import { Entity, ManyToOne, MikroORM, OneToOne, PrimaryKey } from '@mikro-orm/sqlite';

@Entity()
class Breed {
  @PrimaryKey()
  public readonly id!: number;
}

@Entity()
class Pet {
  @PrimaryKey()
  public readonly id!: number;

  @ManyToOne()
  public readonly breed!: Breed;
}

@Entity()
class User {
  @PrimaryKey()
  public readonly id!: number;

  @OneToOne()
  public readonly pet!: Pet;
}

let orm: MikroORM;

beforeAll(async () => {
  orm = await MikroORM.init({
    entities: [User, Pet, Breed],
    dbName: `:memory:`,
    allowGlobalContext: true,
  });
  await orm.schema.refreshDatabase();
});

afterAll(() => orm.close(true));

test('test', async () => {
  const user = orm.em.create(User, { pet: { breed: {} } });
  await orm.em.persistAndFlush([user]);

  const dbUser = await orm.em
    .createQueryBuilder(User)
    .select('*')
    .where({
      pet: {
        breed: {
          id: {
            $ne: null,
          },
        },
      },
    })
    .leftJoinAndSelect('pet', 'p')
    .getSingleResult();

  expect(dbUser?.pet).not.toBeNull();
});
```

### What driver are you using?

@mikro-orm/sqlite

### MikroORM version

6.2.9

### Node.js version

22.4.1

### Operating system

mac

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
If you first do the join, your where condition would reuse the join branch and it works fine:

```ts
  const dbUser = await orm.em
    .createQueryBuilder(User)
    .select('*')
    .leftJoinAndSelect('pet', 'p')
    .where({
      pet: {
        breed: {
          id: {
            $ne: null,
          },
        },
      },
    })
    .getSingleResult();
```

I don't think it makes sense to first apply a deep nested condition and join afterward, joins need to happen first.

**Reply by vinverdy** [reply]
i see, thankyou.
